internal/services/iothub/iothub_endpoint_servicebus_queue_resource.go

```diff
@@ -159,10 +168,18 @@ func resourceIotHubEndpointServiceBusQueueCreateUpdate(d *pluginsdk.ResourceData
 	queueEndpoint := devices.RoutingServiceBusQueueEndpointProperties{
 		AuthenticationType: authenticationType,
 		Name:               utils.String(id.EndpointName),
-		SubscriptionID:     utils.String(subscriptionID),
 		ResourceGroup:      utils.String(endpointRG),
 	}
 
+	// To align with the previous TF behavior, `subscription_id` needs to be set with the provider's subscription Id when it isn't specified in the tf config, otherwise TF behavior is different than before and it may block the existing users
+	// From the business perspective, the raw config handling is only deant for the case that the user has an CosmosDB Account whose Endpoint's subscription is not the provider's one. Then the user wants to reset it to the provider's one by unset the subscription_id
```

Comment:
```suggestion
	// From the business perspective, the raw config handling is only meant for the case that the user has an CosmosDB Account whose Endpoint's subscription is not the provider's one. Then the user wants to reset it to the provider's one by unset the subscription_id
```
Did you mean `meant`?

Reply:
Updated